src/lib/webhook-server.ts

```diff
@@ -0,0 +1,108 @@
+import assert = require("assert");
+import bufferEqualsConstantTime = require("buffer-equals-constant");
+import { createHmac } from "crypto";
+import { createServer, Server } from "http";
+import full from "../full";
+import RollingLogs from "./rolling-logs";
+import { ArrayLog, settings } from "./common";
+import { currentTimeStamp, indent, parseJson } from "./util";
+
+const rollingLogs = new RollingLogs("webhook-logs.md", 1000);
+
+export default function server(key: string, dry: boolean): Server {
+	return listenToGithub(dry, key, updateOneAtATime(async (log, timeStamp) => {
+		log.info(""); log.info("");
+		log.info(`# ${timeStamp}`);
+		log.info("");
+		log.info("Starting full...");
+		try {
+			await full(dry, timeStamp);
+		} catch (err) {
+			log.info("# ERRROR");
+			log.info("");
+			log.info(indent(err.stack));
+		}
+
+		if (!dry) {
+			await writeLog(log);
+		}
+	}));
+}
+
+function writeLog(log: ArrayLog): Promise<void> {
+	const { infos, errors } = log.result();
+	assert(!errors.length);
+	return rollingLogs.write(infos);
+}
+
+function listenToGithub(dry: boolean, key: string, onUpdate: (log: ArrayLog, timeStamp: string) => void): Server {
```

Comment:
Nit if you feel like it: `key` and `dry` should be in the same order as in `server`